pisa-proxy/app/config/src/config.rs

```diff
@@ -21,185 +21,208 @@ use proxy::proxy::{MySQLNode, MySQLNodes, ProxiesConfig, ProxyConfig};
 use serde::{Deserialize, Serialize};
 use tracing::{info, trace};
 
-#[derive(Debug, Deserialize, Serialize)]
-pub struct Config {
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct PisaProxyConfig {
     pub admin: Admin,
-    pub mysql: Option<MySQLNodes>,
     pub proxy: Option<ProxiesConfig>,
+    pub mysql: Option<MySQLNodes>,
+    pub shardingsphere_proxy: Option<MySQLNodes>,
+    pub version: Option<String>,
 }
 
-#[derive(Debug, Clone)]
-pub enum Node {
-    MySQL(Vec<MySQLNode>),
-    ShardingSphereProxy(Vec<MySQLNode>),
-}
-
-#[derive(Debug, Clone, Serialize, Deserialize, Default)]
-pub struct PisaConfig {
-    pub admin: Admin,
-    pub proxies: Vec<ProxyConfig>,
-    pub mysql_nodes: Vec<MySQLNode>,
-    pub shardingsphere_proxy_nodes: Vec<MySQLNode>,
-    pub version: String,
+#[derive(Default, Clone)]
+pub struct PisaProxyConfigBuilder {
+    pub _admin: Admin,
+    pub _proxy: Option<ProxiesConfig>,
+    pub _mysql: Option<MySQLNodes>,
+    pub _shardingsphere_proxy: Option<MySQLNodes>,
+    pub _version: Option<String>, 
+
+    pub _local: String,
+    pub _config_path: String,
+    pub _http_path: String,
+    pub _log_level: String,
+    pub _port: String,
```

Comment:
This should be used `u16` type here.

Reply:
This `_port` is read from comand line arguements, so I set it to String, and converting to u32 when building the PisaProxyConfig entity.